ashlar/reg.py

```diff
@@ -730,22 +746,34 @@ def register_all(self):
             print()
 
     def calculate_positions(self):
-        self.positions = self.tile_positions + self.shifts
+        self.positions = self.corrected_nominal_positions + self.shifts
+        # self.reference_aligner.shifts[reference_idx] does not equal to 
+        # (self.reference_aligner_positions - self.reference_positions)
+        self.positions += (
+            self.reference_aligner_positions - 
+            self.reference_positions
+        )
         self.constrain_positions()
         self.centers = self.positions + self.metadata.size / 2
 
     def constrain_positions(self):
-        # Computed shifts of exactly 0,0 seem to result from failed
-        # registration. We need to throw those out for this purpose.
-        discard = (self.shifts == 0).all(axis=1)
+        # Discard camera background registration which will shift target 
+        # positions to reference aligner positions
```

Comment:
We should explain this a bit more. Maybe add ", due to strong self-correlation of the sensor dark current pattern which dominates in low-signal images."